ProcessLib/PhaseField/PhaseFieldProcess-impl.h

```diff
@@ -179,19 +342,34 @@ void PhaseFieldProcess<DisplacementDim>::preTimestepConcreteProcess(
 
     GlobalExecutor::executeMemberOnDereferenced(
         &LocalAssemblerInterface::preTimestep, _local_assemblers,
-        *_local_to_global_index_map, x, t, dt);
+        getDOFTable(process_id), x, t, dt);
 }
 
 template <int DisplacementDim>
 void PhaseFieldProcess<DisplacementDim>::postTimestepConcreteProcess(
-    GlobalVector const& x, int const /*process_id*/)
+    GlobalVector const& x, int const process_id)
 {
     DBUG("PostTimestep PhaseFieldProcess.");
 
     GlobalExecutor::executeMemberOnDereferenced(
         &LocalAssemblerInterface::postTimestep, _local_assemblers,
-        *_local_to_global_index_map, x);
+        getDOFTable(process_id), x);
 }
 
+template <int DisplacementDim>
+void PhaseFieldProcess<DisplacementDim>::postNonLinearSolverConcreteProcess(
+    GlobalVector const& x, const double t, const int process_id)
+{
+    if (!_use_monolithic_scheme && process_id == 0)
```

Comment:
@wenqing Can this decision be made before calling this function?

Reply:
postNonLinearSolver function is yet to be used in this commit. Thus, the function has been removed.